valhalla/jawn/src/managers/score/ScoreManager.ts

```diff
@@ -87,6 +101,128 @@ export class ScoreManager extends BaseManager {
+            const existingMessage = map.get(key);
+            if (
+              !existingMessage ||
+              existingMessage.createdAt < message.createdAt
+            ) {
+              map.set(key, message);
+            }
+            return map;
+          }, new Map<string, HeliconeScoresMessage>())
+          .values()
+      );
+      const bumpedVersions = await this.scoreStore.bumpRequestVersion(
+        filteredMessages.map((scoresMessage) => ({
+          id: scoresMessage.requestId,
+          organizationId: scoresMessage.organizationId,
+        }))
+      );
+
+      if (bumpedVersions.error || !bumpedVersions.data) {
+        return err(bumpedVersions.error);
+      }
+      const scoresScoreResult = await this.scoreStore.putScoresIntoClickhouse(
+        bumpedVersions.data.map((scoresMessage) => {
+          return {
+            requestId: scoresMessage.id,
+            organizationId: scoresMessage.helicone_org_id,
+            provider: scoresMessage.provider,
+            version: scoresMessage.version,
+            mappedScores:
+              filteredMessages.find((x) => x.requestId === scoresMessage.id)
+                ?.scores ?? [],
+          };
+        })
+      );
+
+      if (scoresScoreResult.error) {
+        console.error("Error upserting scores:", scoresScoreResult.error);
+        return err(scoresScoreResult.error);
+      }
+      return ok(null);
+    } catch (error: any) {
+      console.error("Error processing scores message:", error.message);
+      return err(error.message);
+    }
+  }
+
+  public async handleScores(
+    batchContext: {
+      batchId: string;
+      partition: number;
+      lastOffset: string;
+      messageCount: number;
+    },
+    scoresMessages: HeliconeScoresMessage[]
+  ): Promise<Result<null, string>> {
+    console.log(`Handling scores for batch ${batchContext.batchId}`);
+    const start = performance.now();
+    const result = await this.procesScores(scoresMessages);
+    const end = performance.now();
+    const executionTimeMs = end - start;
+
+    dataDogClient.logHandleResults({
+      executionTimeMs,
+      handlerName: "scoresHandler",
+      methodName: "handlescores",
+      messageCount: batchContext.messageCount,
+      message: "scores",
+    });
+
+    if (result.error) {
+      Sentry.captureException(new Error(JSON.stringify(result.error)), {
+        tags: {
+          type: "HandlescoresError",
+          topic: "helicone-scores-prod",
+        },
+        extra: {
+          batchId: batchContext.batchId,
+          partition: batchContext.partition,
+          offset: batchContext.lastOffset,
+          messageCount: batchContext.messageCount,
+        },
+      });
+
+      console.error(
+        `Error inserting scores: ${JSON.stringify(result.error)} for batch ${
+          batchContext.batchId
+        }`
+      );
+
+      const kafkaProducer = new KafkaProducer();
+      const kafkaResult = await kafkaProducer.sendScoresMessage(
```

Comment:
I dont really understand why are we producing here. Isnt this the consumer? do we mean to produce to the DLQ?

Reply:
Good catch! Yeah should be DLQ there